rust/lancedb/src/table.rs

```diff
@@ -1086,6 +1094,16 @@ impl Table {
         self.inner.drop_index(name).await
     }
 
+    /// Poll until the columns are fully indexed. Will return Error::Timeout if the columns
+    /// are not fully indexed within the timeout.
+    pub async fn wait_for_index(
+        &self,
+        index_names: &[&str],
```

Comment:
One design question is if this should take index names or columns as parameters. Index names seems more appropriate, and consistent with index_stats(), but the "_idx" convention is a bit funky and not very explicit.

Reply:
I agree the convention is funky. In Lance, the name is a parameter users would request.

One thing we should consider is having `create_index` return an object with information about the index, including it's name.

I think for now taking an index name makes the most sense, and we can work on making that name more discoverable for users later.

Reply:
We use the index name for `drop` (I seem to recall a similar concern)